### PR TITLE
fix(web): deduplicate PR project filter choices

### PR DESCRIPTION
**Pull request by juliusmarminge**
The PR Project filter listed every checkout independently, so several checkouts of the same repository on one server appeared as identical choices.

Show one choice per repository per server, preserve saved checkout selections, and distinguish remaining name collisions. Selected options use a shared, opt-in radio indicator. Other menus and merge-readiness sorting/fetching are unchanged.

## Verification

- 152 focused PR list, filter, and assignment tests pass, including 11 project-choice regressions.
- Web typecheck passes; scoped lint has no errors (existing warnings only).
- Browser verified duplicate collapse, selecting/clearing, and a saved URL naming an alternate checkout. Clearing retains the existing server scope.
- Fresh whole-PR audit found no remaining blockers. CI jobs that ran pass, configured skips remain skipped, and the bot finding is fixed and resolved.
- Web and desktop share this route; browser verification used web. Multi-environment cases have unit coverage. No mobile, provider, wire-contract, or documentation changes were needed.

## Before / after

Actual base [7a089b2](https://github.com/pingdotgg/t3code/commit/7a089b2b2449c5c146e1a7d554a31e6d55924d3f) and head [fa907505](https://github.com/pingdotgg/t3code/commit/fa9075053602e8b7285c83aeec4380e094fd2b5e), with the same disposable three-project fixture, query, 1440×1000 viewport, and scroll position.

| Before | After |
| --- | --- |
| Before: three identical repository choices | After: one repository choice and a selection checkmark |

<details>
<summary>Full-page screenshots</summary>

Before:

Before: PR Project filter with duplicate choices

After:

After: deduplicated PR Project filter

</details>

Select and clear, on the final head (light theme):

https://gh-file-drop-api-prod-mi5fy3sowv63ufte.pinglabs.workers.dev/f/fe03ecfbef2c98bc/project-filter-demo-final.mp4

Implemented with vega-alpha in the Codex harness.

---

> [!NOTE]
> **Medium Risk**
> Filter scope now collapses multiple checkouts of the same repo on one server to a single row, which can change which project a user picks unless the saved-selection and suffix logic matches expectations.
> 
> **Overview**
> The PR **Project** filter no longer lists every worktree separately. It now shows **one choice per canonical repository per server**, while still honoring a saved scope when the user had picked a specific checkout.
> 
> A new `pullRequestFilterProjects` helper replaces the inline “duplicate title → append server label” logic on the pull-requests route. It keys rows by environment plus repository identity (case-insensitive), keeps the explicitly selected project when collapsing duplicates, and only adds disambiguating title suffixes (server label, path, environment id, project id) when names would still collide.
> 
> Filter radio menus also show a **checkmark** on the active option via a new `MenuRadioItemIndicator` in the shared menu component, wired into `PullRequestFilterRadioGroup`. Regression tests cover collapse, saved worktree selection, unknown identities, and edge-case naming.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit fa9075053602e8b7285c83aeec4380e094fd2b5e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Deduplicate pull-request project filter choices by repository and environment
> - Adds `pullRequestFilterProjects` in [pullRequestProjectFilter.logic.ts](https://github.com/pingdotgg/t3code/pull/9948/files#diff-da2110bab4d1769f666c7a3d99b7c8320b105915f6ecc20377bf91240625a4a7) that groups projects per environment and lowercased canonical repository identity, collapses duplicate checkouts (retaining the selected one), and disambiguates titles by environment label, checkout path, environment id, then project id
> - Adds `distinguishTitles` utility that appends a suffix only to titles appearing more than once, and a reusable `MenuRadioItemIndicator` in [menu.tsx](https://github.com/pingdotgg/t3code/pull/9948/files#diff-e0c9f33f4f308eae4b7e310785187bc81767e547539de6fcd955fa42aaccc621) so radio options show a check mark on the selected item
> - Updates [PullRequestsRouteView](https://github.com/pingdotgg/t3code/pull/9948/files#diff-514a5b26d9687035b089d834742d874da174aa0a8a0c7c2fd47c61d11795afa5) to call the new utility, replacing route-local title-counting logic, and updates `PullRequestFiltersMenu` input types so optional favicon/icon fields accept `undefined`
> - Adds extensive tests in [pullRequestProjectFilter.logic.test.ts](https://github.com/pingdotgg/t3code/pull/9948/files#diff-8c00ea6804fe15861a507ac6802dc9908b77d4200c74c71039608047efe73de9) covering case-insensitive matching, cross-host distinction, unknown-identity retention, and ordering
> - Behavioral Change: filter choices now show one entry per repository per environment instead of one per project checkout; previously distinct duplicate checkouts are collapsed into a single choice
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized fa90750.</sup>
> 
>

### CODE REVIEW

File: apps/web/src/components/pullRequest/PullRequestListFilters.tsx [hunk 1]
```diff
@@ -205,6 +206,7 @@ function PullRequestFilterRadioGroup<Value extends string>({
               <PullRequestFilterOptionIcon option={option} />
               <span className="min-w-0 flex-1 truncate">{option.label}</span>
               {option.unavailable ? <span className="shrink-0">· Unavailable</span> : null}
+              {option.value === value ? <CheckIcon aria-hidden className="size-3.5" /> : null}
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply apps/web/src/components/pullRequest/PullRequestListFilters.tsx]: Addressed in [fa9075053](https://github.com/pingdotgg/t3code/commit/fa9075053602e8b7285c83aeec4380e094fd2b5e): the shared `MenuRadioItemIndicator` now owns the checkmark and reads Base UI radio state. PR filters opt into that slot; unrelated menus keep their existing appearance. Verified 152 focused tests, web typecheck, and the saved-project selection in the browser.
